bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -875,6 +881,20 @@ public void run() {
     @Override
     public void modifiedService(ServiceReference/* <ModuleHandlerFactory> */ reference,
             /* ModuleHandlerFactory */Object service) {
+        logger.debug("ModuleHandlerFactory modified, updating handlers");
+        ModuleHandlerFactory moduleHandlerFactory = ((ModuleHandlerFactory) service);
+        @SuppressWarnings("unchecked")
+        Collection<String> removed_handlers = (Collection<String>) reference.getProperty("removed_handlers");
```

Comment:
The module type definition can be defined in separate bundle from their handler implementations, but we have discussion and agree that it is unusual case and it's better to be coupled into one bundle. My point of view for the module types and handlers is they are some kind of static information (which is predefined, not defined runtime with Java or JavaScript) which can be dynamically installed  or removed as bundle.
But if you think this feature will be useful, it's ok for me.